boards/arm/mimxrt685_evk/pinmux.c

```diff
@@ -278,6 +278,239 @@ static int mimxrt685_evk_pinmux_init(const struct device *dev)
+IOPCTL_PinMuxSet(IOPCTL, 2U, 17U, port2_pin17_config);
+
+const uint32_t port2_pin18_config = (/* Pin is configured as FLEXSPI0B_DATA5 */
+		IOPCTL_PIO_FUNC6 |
+		/* Disable pull-up / pull-down function */
+		IOPCTL_PIO_PUPD_DI |
+		/* Enable pull-down function */
+		IOPCTL_PIO_PULLDOWN_EN |
+		/* Enables input buffer function */
+		IOPCTL_PIO_INBUF_EN |
+		/* Normal mode */
+		IOPCTL_PIO_SLEW_RATE_NORMAL |
+		/* Full drive enable */
+		IOPCTL_PIO_FULLDRIVE_EN |
+		/* Analog mux is disabled */
+		IOPCTL_PIO_ANAMUX_DI |
+		/* Pseudo Output Drain is disabled */
+		IOPCTL_PIO_PSEDRAIN_DI |
+		/* Input function is not inverted */
+		IOPCTL_PIO_INV_DI);
+/* PORT2 PIN18 (coords: R2) is configured as FLEXSPI0B_DATA5 */
+IOPCTL_PinMuxSet(IOPCTL, 2U, 18U, port2_pin18_config);
+
+const uint32_t port2_pin19_config = (/* Pin is configured as FLEXSPI0B_SS0_N */
+		IOPCTL_PIO_FUNC6 |
+		/* Disable pull-up / pull-down function */
+		IOPCTL_PIO_PUPD_DI |
+		/* Enable pull-down function */
+		IOPCTL_PIO_PULLDOWN_EN |
+		/* Enables input buffer function */
+		IOPCTL_PIO_INBUF_EN |
+		/* Normal mode */
+		IOPCTL_PIO_SLEW_RATE_NORMAL |
+		/* Full drive enable */
+		IOPCTL_PIO_FULLDRIVE_EN |
+		/* Analog mux is disabled */
+		IOPCTL_PIO_ANAMUX_DI |
+		/* Pseudo Output Drain is disabled */
+		IOPCTL_PIO_PSEDRAIN_DI |
+		/* Input function is not inverted */
+		IOPCTL_PIO_INV_DI);
+/* PORT2 PIN19 (coords: T2) is configured as FLEXSPI0B_SS0_N */
+IOPCTL_PinMuxSet(IOPCTL, 2U, 19U, port2_pin19_config);
+
+const uint32_t port2_pin22_config = (/* Pin is configured as FLEXSPI0B_DATA6 */
+		IOPCTL_PIO_FUNC6 |
+		/* Disable pull-up / pull-down function */
+		IOPCTL_PIO_PUPD_DI |
+		/* Enable pull-down function */
+		IOPCTL_PIO_PULLDOWN_EN |
+		/* Enables input buffer function */
+		IOPCTL_PIO_INBUF_EN |
+		/* Normal mode */
+		IOPCTL_PIO_SLEW_RATE_NORMAL |
+		/* Full drive enable */
+		IOPCTL_PIO_FULLDRIVE_EN |
+		/* Analog mux is disabled */
+		IOPCTL_PIO_ANAMUX_DI |
+		/* Pseudo Output Drain is disabled */
+		IOPCTL_PIO_PSEDRAIN_DI |
+		/* Input function is not inverted */
+		IOPCTL_PIO_INV_DI);
+/* PORT2 PIN22 (coords: P3) is configured as FLEXSPI0B_DATA6 */
+IOPCTL_PinMuxSet(IOPCTL, 2U, 22U, port2_pin22_config);
+
+const uint32_t port2_pin23_config = (/* Pin is configured as FLEXSPI0B_DATA7 */
+		IOPCTL_PIO_FUNC6 |
+		/* Disable pull-up / pull-down function */
+		IOPCTL_PIO_PUPD_DI |
+		/* Enable pull-down function */
+		IOPCTL_PIO_PULLDOWN_EN |
+		/* Enables input buffer function */
+		IOPCTL_PIO_INBUF_EN |
+		/* Normal mode */
+		IOPCTL_PIO_SLEW_RATE_NORMAL |
+		/* Full drive enable */
+		IOPCTL_PIO_FULLDRIVE_EN |
+		/* Analog mux is disabled */
+		IOPCTL_PIO_ANAMUX_DI |
+		/* Pseudo Output Drain is disabled */
+		IOPCTL_PIO_PSEDRAIN_DI |
+		/* Input function is not inverted */
+		IOPCTL_PIO_INV_DI);
+/* PORT2 PIN23 (coords: P5) is configured as FLEXSPI0B_DATA7 */
+IOPCTL_PinMuxSet(IOPCTL, 2U, 23U, port2_pin23_config);
+#endif
```

Comment:
It kind of hurts my eyes, to see the same property and comment so many times, it is consistent with the code above though.

Maybe simplify to (untested & copy-pasted)
```c
const uint32_t common_pin_config = (
		/* Disable pull-up / pull-down function */
		IOPCTL_PIO_PUPD_DI |
		/* Enable pull-down function */
		IOPCTL_PIO_PULLDOWN_EN |
		/* Enables input buffer function */
		IOPCTL_PIO_INBUF_EN |
		/* Normal mode */
		IOPCTL_PIO_SLEW_RATE_NORMAL |
		/* Full drive enable */
		IOPCTL_PIO_FULLDRIVE_EN |
		/* Analog mux is disabled */
		IOPCTL_PIO_ANAMUX_DI |
		/* Pseudo Output Drain is disabled */
		IOPCTL_PIO_PSEDRAIN_DI |
		/* Input function is not inverted */
		IOPCTL_PIO_INV_DI);

/* PORT1 PIN11 (coords: L2) is configured as FLEXSPI0B_DATA0 */
IOPCTL_PinMuxSet(IOPCTL, 1U, 11U, common_pin_config | IOPCTL_PIO_FUNC6);
/* PORT1 PIN12 (coords: M2) is configured as FLEXSPI0B_DATA1 */
IOPCTL_PinMuxSet(IOPCTL, 1U, 12U, common_pin_config | IOPCTL_PIO_FUNC6);

/* etc */
```

But ignore me, if this is a policy :)

Reply:
I just tried to keep it consistent with NXP's SDK code.

Reply:
I agree it's a bit overkill, but it is consistent with the rest of the file. This will soon go away anyway when we move the i.MX RT family to device tree pinmuxing.